src/resolvers/message.ts

```diff
@@ -1,25 +1,76 @@
-import { Message, Resolvers } from '../generated/graphql';
+import { Message, MessagesConnection, Resolvers } from '../generated/graphql';
+import { Op, WhereOptions } from 'sequelize';
 
 import { checkAuth } from '../utils/auth';
+import paginateResults from '../utils/pagination';
 
 const resolver: Resolvers = {
   Query: {
     messages: async (
       _,
-      args, {
-        verifyUser,
-        models,
-      },
-    ): Promise<Message[]> => {
+      args,
+      { verifyUser, models },
+    ): Promise<MessagesConnection> => {
       const { Message: messageModel } = models;
       const auth = verifyUser();
       checkAuth(auth);
 
-      return messageModel.findAll({
-        where: {
-          messageId: 1,
-        },
+      const { message, channelId, pageSize = 20, after } = args;
+      let qryMsg: object = {};
+      let qryAfter: object = {};
+      let qryChannelId: object = {};
+
+      if (message) {
+        const msgOpConds: object = {};
+        const findColName = 'text';
+        msgOpConds[findColName] = {
+          [Op.like]: `%${message}%`,
+        };
+        qryMsg = {
+          [Op.or]: msgOpConds,
+        };
+      }
+      if (after) {
+        qryAfter = {
+          createdAt: { [Op.lt]: Number(after) },
+        };
+      }
+      if (channelId) {
+        qryChannelId = {
+          channelId,
+        };
+      }
+
+      const where: WhereOptions = {
+        ...qryMsg,
+        ...qryAfter,
+        ...qryChannelId,
+      };
```

Comment:
Understand what you are trying to do but I hope same logic stays in on area as much as possible rather than being at A, B, C and do D.

If then, you should make another function and make here more concise. It is even much better at testing as I've shared in previous meetups.